src/client/routes/profile/ProfileView.tsx

```diff
@@ -1,45 +1,92 @@
 import * as React from "react";
-import { withRouter } from "react-router-dom";
-import TextField from "@material-ui/core/TextField";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import {TextField, WithStyles, withStyles, createStyles, Theme, Button, Typography} from "@material-ui/core";
 import ApiService from "../../common/ApiService";
 
-// TODO an actual view with some reasonable ui and more fields! email! password! whoa, very much data, so doge
-class ProfileView extends React.Component<any, any> {
-  constructor(props: any) {
+class ProfileView extends React.Component<IProfileViewProps, IProfileViewState> {
+  private submitField: any = React.createRef();
+
+  constructor(props: IProfileViewProps) {
     super(props);
+    this.state = {} as IProfileViewState
+  }
+
+  public async componentDidMount() {
+    const user = await ApiService.read.user();
+    this.setState(user);
   }
 
   public render() {
-    // todo lol login page class
+
     return (
-      <div className="login page">
-        <div className="form">
-          <h3 className="title">What's your nickname?</h3>
-          <form onSubmit={this.onSubmitLogin}>
-            <TextField name="username" label="Username" onChange={this.onInputChange} />
-          </form>
-        </div>
-      </div>
+      <form
+        className={this.props.classes.container}
+        onKeyPress={this.handleEnterKey}
+      >
+        <TextField
+          autoFocus
+          required
+          id="name"
+          label="Name"
+          className={this.props.classes.textField}
+          value={this.state.name || ""}

```

Comment:
Would it be easier to init this to state in constructor? This way we can add the fields to the state interface as well.

Reply:
State interface already extends IUserContract so it has name among its fields. Personally I don't know which is better, imho both are ugly boilerplating and definciens in the binding engine. I suppose it's easier to read in constructor but meh...